src/schema/v2/me/update_me_mutation.ts

```diff
@@ -190,18 +199,29 @@ export default mutationWithClientMutationId<any, any, ResolverContext>({
     },
     me: Me,
   },
-  mutateAndGetPayload: async (args, { updateMeLoader }) => {
+  mutateAndGetPayload: async (
+    { iconUrl, ...args },
```

Comment:
**thought:** I know this would make things more complicated but is there a particular reason why we are not using the internalID to update the profile icon? (assuming that we get an internalID for the image from Eigen)

Reply:
@MounirDhahri 
1. For a vast majority of users there will be no icon initially so there will be no id to use. Since icon can only be 1 per user at any time, they will be making PUT request to change icon whether they have an icon initially or not.
2. In Eigen, We will be getting the external image url uploaded to s3. Real gemini entries will not be created, neither will icon so we cannot get icon id by uploading image. Gravity will create the entries on Gemini and update the icon after processing via webhook. So from eigen we cannot really do this.
3. Icon is an embedded document in CollectorProfile. Even if we have the icon id we cannot actually do `CollectorProfileIcon.find(id).update({})` because we cannot find the icon by their id. We can access it via `user.collector_profile.icon`. So icon id isn't useful for us here.

Reply:
oki, I see now, thanks for explaining